### PR TITLE
curve

### PR DESCRIPTION
**Pull request by waynebruce0x**

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/curve: 
        
 
  
    
    
    Error: balanceOf multicall failed
    Truncated error stack:
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/generalUtil.js:64:19
        at Array.map (<anonymous>)
        at unwrapPools (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/curve/index.js:320:12)
        at runMicrotasks (<anonymous>)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/curve/index.js:336:7
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:58:25)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:160:9
        at async Promise.all (index 0)

**Comment by llamatester** [comment]
Error while running adapter at projects/curve: 
        
 
  
    
    
    ReferenceError: chain is not defined
    Truncated error stack:
    
        at unwrapSdTokens (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/curve/index.js:246:26)
        at runMicrotasks (<anonymous>)
        at async fixWrappedTokenBalances (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/curve/index.js:126:5)
        at async unwrapPools (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/curve/index.js:322:3)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/curve/index.js:336:7
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:58:25)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:160:9
        at async Promise.all (index 0)
        at async Promise.all (index 0)

**Comment by llamatester** [comment]
The adapter at projects/curve exports TVL: 
        
 
 
    ethereum-staking          337.59 M
    moonbeam                  0
    optimism                  16.21 M
    harmony                   0
    aurora                    28.20 k
    avax                      127.41 M
    arbitrum                  115.86 M
    fantom                    146.05 M
    polygon                   133.25 M
    xdai                      67.52 M
    ethereum                  4.39 B
    staking                   337.59 M
    
    total                    4.99 B

**Comment by llamatester** [comment]
The adapter at projects/curve exports TVL: 
        
 
 
    moonbeam                  0
    ethereum-staking          389.53 M
    optimism                  15.37 M
    aurora                    28.24 k
    avax                      125.02 M
    arbitrum                  107.53 M
    harmony                   0
    fantom                    139.86 M
    xdai                      63.19 M
    polygon                   127.22 M
    ethereum                  4.20 B
    staking                   389.53 M
    
    total                    4.78 B

**Review by g1nt0ki — APPROVED** [review]

**Comment by 0xngmi** [comment]
lfg?

File: projects/curve/index.js [hunk 1]
```diff
@@ -1,100 +1,352 @@
...
+          retry(
+            async bail =>
+              await axios.get(
+                `https://api.exchangerate.host/convert?from=${token.currency}&to=USD`
+              )
+          ),
+          sdk.api.erc20.decimals(token.address, chain)
+        ]);
+
+        sdk.util.sumSingleBalance(
```
Comment (0xngmi) [inline projects/curve/index.js:254]: would be better to use usdc address here so it shows up with the proper symbol

**Comment by 0xngmi** [comment]
@waynebruce0x I reverted commit 9d2e4a7d526f02a22e5301034fe63b671714e239, is this equivalent to merging this PR or is there something i am missing?

**Comment by 0xngmi** [comment]
already merged

**Pull request closed without merging**